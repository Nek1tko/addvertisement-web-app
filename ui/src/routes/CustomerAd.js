import React, { useState } from "react";
import { Box, Typography } from "@mui/material";
import AwesomeSlider from "react-awesome-slider";
import AwesomeSliderStyles from "react-awesome-slider/src/styles";
import eren from "../img/eren.jpg";
import mikasa from "../img/mikasa.jpg";
import levi from "../img/levi.png";
import TextField from "@material-ui/core/TextField";

const CustomerAd = props => {
    const ad = (props.location && props.location.ad) || {};

    const [description, setDescription] = useState(ad.description);
    const [name, setName] = useState(ad.saler.name);
    const [phone, setPhone] = useState(ad.saler.phoneNumber);
    const [adName, setAdName] = useState(ad.name);
    const [price, setPrice] = useState(ad.price);
    const [metro, setMetro] = useState(ad.metro);
    const [category, setCategory] = useState(ad.subcategory.category);
    const [subcategory, setSubcategory] = useState(ad.subcategory);

    return (<div style={{ display: "flex" }}>
        <Box sx={{ width: 1 / 2, height: 1 / 2, flex: 1 }}>
            <Typography variant="h5" align="center" style={{ color: "#666666", marginTop: 30 }}>
                {adName}
            </Typography>

            <Typography align="center" style={{ color: "#666666", marginTop: 10, marginBottom: 10 }}>
                {metro.name}
            </Typography>

            <AwesomeSlider
                animation="foldOutAnimation"
                cssModule={AwesomeSliderStyles}
            >
                <div>
                    <img alt="Ad preview" src={eren} style={{ height: '100%', width: '100%', objectFit: 'contain' }} />
                </div>
                <div>
                    <img alt="Ad preview" src={mikasa}
                        style={{ height: '100%', width: '100%', objectFit: 'contain' }} />
                </div>
                <div>
                    <img alt="Ad preview" src={levi} style={{ height: '100%', width: '100%', objectFit: 'contain' }} />
                </div>
            </AwesomeSlider>
        </Box>

        <Box sx={{ width: 1 / 2, height: 1 / 2, flex: 1 }}>
            <Typography variant="h5" align="center" style={{ color: "#666666", marginTop: 30 }}>
                Цена: {price}
            </Typography>

            <Typography align="center" style={{ color: "#666666", marginTop: 10, marginBottom: 10 }}>
                {category.name}: {subcategory.name}
            </Typography>

            <Typography
                variant="h6"
                align="center"
                style={{ color: "#666666", marginTop: 10, marginBottom: 10 }}
            >
                {name}: {phone}
            </Typography>

            <TextField
                label="Описание"
                variant="filled"
                multiline
                maxRows={11}
                minRows={11}
                value={description}
                fullWidth
                disabled={true}
            />
        </Box>
    </div>
    );
};

export default CustomerAd;