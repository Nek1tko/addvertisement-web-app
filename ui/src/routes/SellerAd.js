import React, {useState} from 'react';
import AwesomeSlider from 'react-awesome-slider';
import AwesomeSliderStyles from 'react-awesome-slider/src/styles';
import eren from '../img/eren.jpg'
import mikasa from '../img/mikasa.jpg'
import levi from '../img/levi.png'
import {Alert, Box, Typography} from "@mui/material";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import {Collapse} from "@material-ui/core";
import DialogActions from "@mui/material/DialogActions";
import Dialog from "@mui/material/Dialog";


const SellerAd = props => {
    const [description, setDescription] = useState('');
    const [open, setOpen] = useState(false);
    const [errorOpen, setErrorOpen] = useState(false);
    const [errorMessage, setErrorMessage] = useState('');
    const [editedDescription, setEditedDescription] = useState(description);
    const [price, setPrice] = useState('');

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleCancelClose = () => {
        setOpen(false);
        setErrorOpen(false);
        setEditedDescription(description);
    };

    const handleSaveClose = () => {
        setOpen(false);
        setErrorOpen(false);
        setDescription(editedDescription);
    };

    return (
        <div style={{display: "flex"}}>
            <Box sx={{width: 1 / 2, height: 1 / 2, flex: 1}}>
                <Typography variant="h5" align="center" style={{color: "#666666", marginTop: 30, marginBottom: 30}}>
                    Название объявления
                </Typography>
                <AwesomeSlider
                    animation="foldOutAnimation"
                    cssModule={AwesomeSliderStyles}
                >
                    <div>
                        <img alt="Ad preview" src={eren} style={{height: '100%', width: '100%', objectFit: 'contain'}}/>
                    </div>
                    <div>
                        <img alt="Ad preview" src={mikasa}
                             style={{height: '100%', width: '100%', objectFit: 'contain'}}/>
                    </div>
                    <div>
                        <img alt="Ad preview" src={levi} style={{height: '100%', width: '100%', objectFit: 'contain'}}/>
                    </div>
                </AwesomeSlider>
            </Box>
            <Box sx={{width: 1 / 2, height: 1 / 2, flex: 1}}>
                <Typography variant="h5" align="center" style={{color: "#666666", marginTop: 30, marginBottom: 30}}>
                    Цена: {price}
                </Typography>
                <TextField
                    label="Описание"
                    variant="filled"
                    multiline
                    maxRows={14}
                    minRows={14}
                    onChange={e => {
                        setDescription(e.target.value);
                    }}
                    value={description}
                    fullWidth
                    disabled={true}
                />
                <Button
                    fullWidth
                    variant="outlined"
                    onClick={handleClickOpen}
                >
                    Редактировать
                </Button>
                <Dialog
                    open={open}
                    onClose={handleCancelClose}
                    fullWidth
                    maxWidth="md"
                >
                    <DialogTitle>Редактирование описания</DialogTitle>

                    <DialogContent>
                        <DialogContentText>
                            Введите новое описание
                        </DialogContentText>

                        <Collapse in={errorOpen}>
                            <Alert
                                severity="error"
                                sx={{mb: 2}}
                            >
                                {errorMessage}
                            </Alert>
                        </Collapse>

                        <TextField
                            label="Описание"
                            variant="filled"
                            value={editedDescription}
                            multiline
                            maxRows={9}
                            minRows={9}
                            onChange={e => {
                                const value = e.target.value;
                                if (value.length < 1 || value.length > 250) {
                                    setErrorOpen(true);
                                    setErrorMessage("Длина описания не должна быть меньше 4 и больше 250");
                                } else {
                                    setEditedDescription(value);
                                    setErrorOpen(false);
                                }
                            }}
                            fullWidth
                        />
                    </DialogContent>

                    <DialogActions>
                        <Button onClick={handleCancelClose}>Отмена</Button>
                        <Button onClick={handleSaveClose}>Сохранить</Button>
                    </DialogActions>
                </Dialog>
            </Box>
        </div>
    )
};

export default SellerAd;