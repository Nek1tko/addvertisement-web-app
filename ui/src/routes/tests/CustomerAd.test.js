import axios from "axios";
import React from "react";
import {mount} from "enzyme";
import CustomerAd from "../CustomerAd";
import {act} from "react-dom/test-utils";
import SellerAd from "../SellerAd";

jest.mock('axios');

const mockedResponse = {
    data: [
        {
            id: 1,
            name: "mock name"
        },
        {
            id: 2,
            name: "mock name"
        },
    ]
};

const mCategory = {
    id: 1,
    name: "mock category"
};

const mSubcategory = {
    id: 1,
    name: "mock subcategory",
    category: mCategory
};

const mMetro = {
    id: 1,
    name: "Улица Дыбенко",
};

const mAd = {
    id: 1,
    description: "mock description",
    price: "123",
    name: "mock name",
    saler: {
        name: "mock name",
    },
    metro: mMetro,
    subcategory: mSubcategory,
    is_active: true,
    isFavourite: true,
};

const props = {
    location: {
        ad: mAd
    }
};

describe('CustomerAdTests', () => {
    let wrapper;
    const setState = jest.fn();

    beforeEach(() => {
        axios.get.mockResolvedValue(mockedResponse);
        axios.put.mockResolvedValue(mockedResponse);
        const useStateSpy = jest.spyOn(React, 'useState');
        useStateSpy.mockImplementation((state) => [state, setState]);
        wrapper = mount(<CustomerAd{...props}/>);
    })

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('CustomerAdRenderTest', () => {
        act(() => {
            mount(<SellerAd {...props}/>);
        });
    });

    test('CustomerAdSnapshotTest', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('CustomerAdFavButtonTest', async () => {
        await act(async () => {
            await wrapper.find('#favButton').first().props().onClick();
        });
        expect(setState).toHaveBeenCalledTimes(0);
    });
});