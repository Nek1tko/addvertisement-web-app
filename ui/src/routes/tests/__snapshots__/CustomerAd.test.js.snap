// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerAdTests CustomerAdSnapshotTest 1`] = `ReactWrapper {}`;
